### PR TITLE
perf: Cherry-pick Lodash imports for smaller bundles

### PR DESCRIPTION
**Pull request by jrood**
### Overview

Use Cherry-picked Lodash imports for smaller bundles

### PR Checklist

- [ ] ~Related to designs:~
- [ ] ~Related to JIRA ticket: [ABC-123]~
- [x] I have run this code to verify it works
- [ ] ~This PR includes unit tests for the code change~
- [ ] ~This PR includes testing instructions tests for the code change~
- [x] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/37591)  | [Monolith Env](https://pr-37591-monolith.dev-eks.codecademy.com/) |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/6046)  | [Portal Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-6046)   |

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.17.7-alpha.aa13ed.0<br/>@codecademy/gamut-icons@9.24.2-alpha.aa13ed.0<br/>@codecademy/gamut-illustrations@0.45.1-alpha.aa13ed.0<br/>@codecademy/gamut-kit@0.6.404-alpha.aa13ed.0<br/>@codecademy/gamut-patterns@0.9.12-alpha.aa13ed.0<br/>@codecademy/gamut-styles@16.2.1-alpha.aa13ed.0<br/>@codecademy/gamut-tests@5.0.10-alpha.aa13ed.0<br/>@codecademy/styleguide@66.18.2-alpha.aa13ed.0<br/>@codecademy/variance@0.21.3-alpha.aa13ed.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://662fc037ca589405bef100f1--gamut-preview.netlify.app>https://662fc037ca589405bef100f1--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/662fc037ca589405bef100f1>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://662fc26c92f2c04207b019b8--gamut-preview.netlify.app>https://662fc26c92f2c04207b019b8--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/662fc26c92f2c04207b019b8>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.17.7-alpha.817a81.0<br/>@codecademy/gamut-icons@9.24.2-alpha.817a81.0<br/>@codecademy/gamut-illustrations@0.45.1-alpha.817a81.0<br/>@codecademy/gamut-kit@0.6.404-alpha.817a81.0<br/>@codecademy/gamut-patterns@0.9.12-alpha.817a81.0<br/>@codecademy/gamut-styles@16.2.1-alpha.817a81.0<br/>@codecademy/gamut-tests@5.0.10-alpha.817a81.0<br/>@codecademy/styleguide@66.18.2-alpha.817a81.0<br/>@codecademy/variance@0.21.3-alpha.817a81.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.17.7-alpha.e14546.0<br/>@codecademy/gamut-icons@9.24.2-alpha.e14546.0<br/>@codecademy/gamut-illustrations@0.45.1-alpha.e14546.0<br/>@codecademy/gamut-kit@0.6.404-alpha.e14546.0<br/>@codecademy/gamut-patterns@0.9.12-alpha.e14546.0<br/>@codecademy/gamut-styles@16.2.1-alpha.e14546.0<br/>@codecademy/gamut-tests@5.0.10-alpha.e14546.0<br/>@codecademy/styleguide@66.18.2-alpha.e14546.0<br/>@codecademy/variance@0.21.3-alpha.e14546.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://662fd1f1e647894a68ab66f0--gamut-preview.netlify.app>https://662fd1f1e647894a68ab66f0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/662fd1f1e647894a68ab66f0>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://662fd4f4e647894e52ab633e--gamut-preview.netlify.app>https://662fd4f4e647894e52ab633e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/662fd4f4e647894e52ab633e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.17.7-alpha.23ec48.0<br/>@codecademy/gamut-icons@9.24.2-alpha.23ec48.0<br/>@codecademy/gamut-illustrations@0.45.1-alpha.23ec48.0<br/>@codecademy/gamut-kit@0.6.404-alpha.23ec48.0<br/>@codecademy/gamut-patterns@0.9.12-alpha.23ec48.0<br/>@codecademy/gamut-styles@16.2.1-alpha.23ec48.0<br/>@codecademy/gamut-tests@5.0.10-alpha.23ec48.0<br/>@codecademy/styleguide@66.18.2-alpha.23ec48.0<br/>@codecademy/variance@0.21.3-alpha.23ec48.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6632528ed6b8e0f019f4cee4--gamut-preview.netlify.app>https://6632528ed6b8e0f019f4cee4--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6632528ed6b8e0f019f4cee4>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.18.1-alpha.0071d2.0<br/>@codecademy/gamut-icons@9.24.2-alpha.0071d2.0<br/>@codecademy/gamut-illustrations@0.45.1-alpha.0071d2.0<br/>@codecademy/gamut-kit@0.6.405-alpha.0071d2.0<br/>@codecademy/gamut-patterns@0.9.12-alpha.0071d2.0<br/>@codecademy/gamut-styles@16.2.1-alpha.0071d2.0<br/>@codecademy/gamut-tests@5.0.10-alpha.0071d2.0<br/>@codecademy/styleguide@66.19.1-alpha.0071d2.0<br/>@codecademy/variance@0.21.3-alpha.0071d2.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66328622ab23be22f45c161d--gamut-preview.netlify.app>https://66328622ab23be22f45c161d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66328622ab23be22f45c161d>Deploy Logs</a></p>

**Review by jakemhiller — COMMENTED** [review]
1. I love this
2. what do you think about adding `eslint-plugin-lodash` and linting this, so it doesn't get reversed in the future?

in the root `.eslintrc.json`
```
{     
      ...
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "plugins": ["lodash"],
      "rules": {
        "lodash/import-scope": ["error", "method-package"],
        ...
```

**Review by dreamwasp — APPROVED** [review]
🥲

**Comment by jrood** [comment]
> 1. I love this
> 2. what do you think about adding `eslint-plugin-lodash` and linting this, so it doesn't get reversed in the future?
> 
> in the root `.eslintrc.json`
> 
> ```
> {     
>       ...
>       "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
>       "plugins": ["lodash"],
>       "rules": {
>         "lodash/import-scope": ["error", "method-package"],
>         ...
> ```

ooh, yes I'll add that!

File: .eslintrc.js [hunk 1]
```diff
@@ -48,5 +48,12 @@ module.exports = {
         '@typescript-eslint/restrict-template-expressions': 'off',
       },
     },
+    {
+      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
+      "plugins": ["lodash"],
+      "rules": {
+        "lodash/import-scope": ["error", "method"],
```
Comment (jrood) [inline .eslintrc.js:55]: I used `method` rather than `method-package` here for `lodash/map` rather than `lodash.map`

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6632a0741163b50c53571589--gamut-preview.netlify.app>https://6632a0741163b50c53571589--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6632a0741163b50c53571589>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.18.1-alpha.f1f5df.0<br/>@codecademy/gamut-icons@9.24.2-alpha.f1f5df.0<br/>@codecademy/gamut-illustrations@0.45.1-alpha.f1f5df.0<br/>@codecademy/gamut-kit@0.6.405-alpha.f1f5df.0<br/>@codecademy/gamut-patterns@0.9.12-alpha.f1f5df.0<br/>@codecademy/gamut-styles@16.2.1-alpha.f1f5df.0<br/>@codecademy/gamut-tests@5.0.10-alpha.f1f5df.0<br/>@codecademy/styleguide@66.19.1-alpha.f1f5df.0<br/>@codecademy/variance@0.21.3-alpha.f1f5df.0<br/></pre>

File: .eslintrc.js [hunk 2]
```diff
@@ -48,5 +48,12 @@ module.exports = {
         '@typescript-eslint/restrict-template-expressions': 'off',
       },
     },
+    {
+      files: ['*.ts', '*.tsx', '*.js', '*.jsx'],
+      plugins: ['lodash'],
+      rules: {
+        'lodash/import-scope': ['error', 'method'],
```
Comment (jrood) [inline .eslintrc.js:55]: I used `method` rather than `method-package` here for `lodash/map` rather than `lodash.map`

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.18.1-alpha.4bbac1.0<br/>@codecademy/gamut-icons@9.24.2-alpha.4bbac1.0<br/>@codecademy/gamut-illustrations@0.45.1-alpha.4bbac1.0<br/>@codecademy/gamut-kit@0.6.405-alpha.4bbac1.0<br/>@codecademy/gamut-patterns@0.9.12-alpha.4bbac1.0<br/>@codecademy/gamut-styles@16.2.1-alpha.4bbac1.0<br/>@codecademy/gamut-tests@5.0.10-alpha.4bbac1.0<br/>@codecademy/styleguide@66.19.1-alpha.4bbac1.0<br/>@codecademy/variance@0.21.3-alpha.4bbac1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6632a3b5cfe16b0f122f7c7c--gamut-preview.netlify.app>https://6632a3b5cfe16b0f122f7c7c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6632a3b5cfe16b0f122f7c7c>Deploy Logs</a></p>

**Pull request merged**